docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+# Proposal B - Extend Platform Variants and Introduce Compatibility Artifact
+
+This proposal recommends changes to image spec for more granular image selection and introduces a new image compatibility artifact type.
+Based on the artifact, cluster administrators can verify hosts or prepare the appropriate infrastructure for containers.
+
+## Modifications
+
+### Image Spec
+
+Extend the platform variants with new ARM and AMD64 micro-architecture levels.
+
+| ISA/ABI          | architecture | variant |
+|------------------|--------------|---------|
+| ...              | ...          | ...     |
+| ARM 64-bit, v9   | arm          | v9      |
+| AMD64 64-bit, v2 | amd64        | v2      |
+| AMD64 64-bit, v3 | amd64        | v3      |
+| AMD64 64-bit, v4 | amd64        | v4      |
+
+## Image Compatibility Artifact
+
+!["Image Compatibility Artifact"](img/B_artifact.png "Image Compatibility Artifact")
+
+### Definitions
+
+* image-compatibility artifact - an OCI artifact that stores the compatibility spec.
+* Compatibility Spec - a json object that represents a graph of namespaces.
+* Namespace - a validation area under which users can describe plugin specs that are consumed by plugins (executable binaries) as input.
+* Plugin - an executable binary that extracts data and checks compatibility on a host.
+
+A new imageCompatibility artifact type that contains a layer with a JSON compatibility object.
+The object represents a graph of namespaces.
+Namespace describes the validation area and is another object containing data that provides plugins specification for the corresponding plugins.
+A plugin is an executable binary that validates the host and returns information about success or failure.
+
+### Compatibility Spec
+
+Compatibility spec is a graph that can have multiple roots with different paths.
+If one namespace in the path fails, the path is incompatible with the host.
+Namespace fails if at least one plugin reports an error.
+If all paths fail, the host is not compatible with the container.
+
+The host is compatible with the container if at least one path is completed successfully.
+
+#### Schema
+
+- **`schemaVersion`** *int*
```

Comment:
should the semver be used here. especially it's meaning full to use "0.1.0" or "0.1.0-draft", otherwise we can not make sure the compatibility spec released is formal verison or draft version, and which OCI spec release  followed

Reply:
Fixed.